src/js/aos.js

```
@@ -16,6 +16,7 @@ import detect from './helpers/detector';
 import handleScroll from './helpers/handleScroll';
 import prepare from './helpers/prepare';
 import elements from './helpers/elements';
+import validContainer from './helpers/container';
```

This function should probably be called “validateContainer” as this form describes the action being takes. On the other hand this not only validates the container but also returns the valid one. Maybe “getContainer” that internally uses “validateContainer” would be a better fit. What do you think?

---

See my comment at https://github.com/michalsnik/aos/pull/393#discussion_r222465362

Also, I think it would be better in this case to keep error throwing at the root, instead of nested in an helper.

---

The reason I proposed `getContainer` is that the current function is not void, and in fact it returns resolved container - which might be a bit misleading.
Now that I think about it once again, I think we can better rename it to `resolveContainer` as it seems more natural - and then below if there is no `container` we can throw an Error at the root.